Adding README-SAMPLES.md and removing exiv2samples.1

clanmills opened the pull request:
Fix #912 Let's not spend a lots of time reviewing this file.

clanmills committed:
Adding README-SAMPLES.md and removing exiv2samples.1

piponazo committed:
Remove samples man page from CMake

piponazo reviewed (dismissed):
As you proposed, I did not spend much time reviewing the new document. I assume that it is just a conversion from the previous man pages, and I trust in your good judgement.

I added an additional commit to fix the CMake. We were installing the man page file that you removed, and therefore I removed that file from the `install()` CMake commands.

clanmills commented:
Well spotted about the INSTALL.  I thought of that, did a grep, and surprised not to find it.  I also thought there was a reference to exiv2samples from the exiv2 man page (src/exiv2.1) and couldn't find that either!

Thanks very much.  Let's push this into v0.27.2 and 'master'.

D4N commented on README-SAMPLES.md:
````diff
@@ -0,0 +1,502 @@
+```
+Usage: ini-test
+```
+This program is used to test reading the file ini-test.  This program was added in Exiv2 v0.26 when the ~/.exiv2 file was added to the Exiv2 architecture.  Please read the code for further information.
````
```suggestion
This program is used to test reading the file ini-test.  This program was added in Exiv2 v0.26 when the ~/.exiv2 file was added to the Exiv2 architecture.
```

clanmills replied:
Changed.

D4N commented on README-SAMPLES.md:
````diff
@@ -0,0 +1,502 @@
+Commands read from stdin.
+```
+
+Sample program test the Iptc reading and writing.
````
```suggestion
Sample program to test IPTC reading and writing.
```

clanmills replied:
Changed

D4N reviewed (changes requested):
The page is not properly rendered on GitHub: https://github.com/Exiv2/exiv2/blob/fix912_README_SAMPLES.md/README-SAMPLES.md
All the headings don't appear as headings but as verbatim ### instead.

clanmills commented:
Thanks for pointing out that GitHub Markdown processes `###` in a different way from MacDown.  I've left blank lines before/after the `###` headings and I hope that'll be effective as that worked in the other markdown documents I have authored.

clanmills committed:
Fixes following review by Dan.

clanmills committed:
Fixing a couple of typos (Usage: Usage:)

clanmills committed:
Add all sample programs to Doxygen.

clanmills committed:
Adding links to source code on exiv2.org

- Updated following review. Uses source URLs on github instead of exiv2.org.
- Change following review.  I'm not convinced we need this, nor do I really
  understand this.  I think a markdown file should be 100% self contained.
  On MacOS-X, this file now opens the file in Xcode, whereas the URL opened
  the Safari Browser.  Anyway, I've made the change.
- Polished to remove typos (with Apologies to Luis who has already approved this change).

piponazo commented on README-SAMPLES.md:
```diff
@@ -10,23 +10,23 @@ Exiv2 is a C++ library and a command line utility to read, write, delete and mod
-| _**xmpsample**_   | Demonstrates Exiv2 library high level XMP classes | [xmpsample](#xmpsample)<br>[https://exiv2.org/examples.html#example5](https://exiv2.org/examples.html#example5) |
+| Name | Purpose | More information | Code |
+|:---  |:---  |:---              |:-- |
+| _**addmoddel**_   | Demonstrates Exiv2 library APIs to add/ modify/ delete metadata | [addmoddel](#addmoddel) | [addmoddel.cpp](https://exiv2.org/doc/addmoddel_8cpp-example.html) |
```
Hum ... I am not sure about this change. Instead of pointing to the exiv2.org doxygen generated documentation, I would point directly to the source file in the repository. In that way, we are sure that users will see the source file corresponding to each branch.

clanmills replied:
That's a good idea.  I didn't know that every file on GitHub is beautifully formatted in colour with line numbers (no line numbers from doxygen formatted code).

I'll chose the 'master' branch. (and not 0.27-maintenance).  These twelve samples (exifprint, geotag, etc) are seldom changed.  There's a little pull-down for different branches.

https://github.com/Exiv2/exiv2/blob/master/src/actions.cpp

piponazo replied:
I was thinking about a relative path, so that you do not need to point to a specific branch:
https://stackoverflow.com/questions/7653483/github-relative-link-in-markdown-file

I know it works when pointing to other .md files, but I am not sure if it will work when pointing to other kind of files such as .cpp.

piponazo commented:
I rebased the branch in top of 0.27-maintenance and squashed the latest commits which were making changes over the links. Please @D4N let us know if you are happy with this version

D4N reviewed (approved):
LGTM, thanks for all these changes. Especially the direct links to the source on GitHub are a really neat idea!
